fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ReviewQualityCheckBottomSheet.kt

```diff
@@ -0,0 +1,92 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import mozilla.components.lib.state.ext.observeAsState
+import org.mozilla.fenix.shopping.state.ReviewQualityCheckAction
+import org.mozilla.fenix.shopping.state.ReviewQualityCheckState
+import org.mozilla.fenix.shopping.state.ReviewQualityCheckState.OptedIn.ProductReviewState.ProductAnalysis
+import org.mozilla.fenix.shopping.state.ReviewQualityCheckStore
+
+/**
+ * Top-level UI for the Review Quality Check feature.
+ *
+ * @param store [ReviewQualityCheckStore] that holds the state.
+ * @param onRequestDismiss Invoked when a user action requests dismissal of the bottom sheet.
+ * @param modifier The modifier to be applied to the Composable.
+ */
+@Composable
+fun ReviewQualityCheckBottomSheet(
+    store: ReviewQualityCheckStore,
+    onRequestDismiss: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val reviewQualityCheckState by store.observeAsState(ReviewQualityCheckState.Initial) { it }
+
+    ReviewQualityCheckScaffold(
+        onRequestDismiss = onRequestDismiss,
+        modifier = modifier.animateContentSize(),
+    ) {
+        when (val state = reviewQualityCheckState) {
+            is ReviewQualityCheckState.NotOptedIn -> {
+                ReviewQualityCheckContextualOnboarding(
+                    onPrimaryButtonClick = {
+                        store.dispatch(ReviewQualityCheckAction.OptIn)
+                    },
+                    onSecondaryButtonClick = onRequestDismiss,
+                )
+            }
+
+            is ReviewQualityCheckState.OptedIn -> {
+                ProductReviewState(
+                    state = state,
+                    onOptOutClick = {
+                        onRequestDismiss()
+                        store.dispatch(ReviewQualityCheckAction.OptOut)
+                    },
+                    onProductRecommendationsEnabledStateChange = {
+                        store.dispatch(ReviewQualityCheckAction.ToggleProductRecommendation)
+                    },
+                )
+            }
+
+            is ReviewQualityCheckState.Initial -> {}
+        }
+    }
+}
+
+@Composable
+private fun ProductReviewState(
+    state: ReviewQualityCheckState.OptedIn,
+    onOptOutClick: () -> Unit,
+    onProductRecommendationsEnabledStateChange: (Boolean) -> Unit,
+) {
+    when (val productReviewState = state.productReviewState) {
+        is ProductAnalysis -> {
+            ProductAnalysis(
+                productRecommendationsPreference = state.productRecommendationsPreference,
+                productAnalysis = productReviewState,
+                onOptOutClick = onOptOutClick,
+                onProductRecommendationsEnabledStateChange = onProductRecommendationsEnabledStateChange,
+            )
+        }
+
+        is ReviewQualityCheckState.OptedIn.ProductReviewState.Error -> {
+            // Bug 1840113
```

Comment:
nit: the bugs + comments aren't super necessary. we know it's all WIP

Reply:
This will help us work on it in parallel, I don't think this hurts, a placeholder till we add the implementations here